### PR TITLE
[#2927] Improvement(catalog-lakehouse-iceberg): Support more file formats in using clause when create iceberg tables

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?
Support more file formats in using clause when create iceberg tables, such as `using parquet`, `using orc`, `using avro`, `using iceberg`.

using other provider will throw an exception.

### Why are the changes needed?
Because Iceberg official supports using parquet/orc/avro/iceberg when create a new table.

Fix: https://github.com/datastrato/gravitino/issues/2927

### Does this PR introduce _any_ user-facing change?
Yes, users can using `parquet/orc/avro/iceberg` keywork in `using clause` when create a new table.

### How was this patch tested?
New IT.

### CODE REVIEW

**Comment by caican00** [comment]
Hi @FANNG1 could you help review this PR when you are free? Thanks

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java [hunk 1]
```diff
@@ -47,12 +48,32 @@ public class IcebergTable extends BaseTable {
   /** The default provider of the table. */
   public static final String DEFAULT_ICEBERG_PROVIDER = "iceberg";
 
+  /** The supported file formats for Iceberg tables. */
+  public static final String ICEBERG_PARQUET_FILE_FORMAT = "parquet";
+
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java:53]: remove blank line?

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java [hunk 2]
```diff
@@ -1051,6 +1052,114 @@ public void testTableDistribution() {
...
+            table_comment,
+            properties,
+            partitioning,
+            distribution,
+            sortOrders);
+    Assertions.assertFalse(createdTable.properties().containsKey(DEFAULT_FILE_FORMAT));
+    loadTable = tableCatalog.loadTable(tableIdentifier);
+    Assertions.assertFalse(loadTable.properties().containsKey(DEFAULT_FILE_FORMAT));
+
+    tableCatalog.dropTable(tableIdentifier);
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java:1103]: 1. use `ICEBERG_ORC_FILE_FORMAT ` not `orc` directly?
2. do some abstraction to simple the code?

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java (hunk 1 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java:53]: > remove blank line?

When spotlessApply is executed, empty lines are automatically added.
Maybe we can add comments to each format？

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java (hunk 2 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java:1103]: ok.

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java (hunk 1 above)
Comment (FANNG1) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java:53]: both ok for me

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java (hunk 2 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java]: updated.

**Comment by caican00** [comment]
all comments have been addressed.

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java [hunk 3]
```diff
@@ -1051,6 +1057,87 @@ public void testTableDistribution() {
...
+                tableIdentifier,
+                columns,
+                table_comment,
+                properties,
+                partitioning,
+                distribution,
+                sortOrders));
+  }
+
+  private static void assertionsTableProperties(
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java:1123]: rename to checkIcebergTableFileFormat?

**Comment by FANNG1** [comment]
LGTM, except one minor comment

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java (hunk 3 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java:1123]: ok

Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java]: updated

**Comment by caican00** [comment]
> LGTM, except one minor comment

new comments have been addressed.

**Comment by FANNG1** [comment]
> > LGTM, except one minor comment
> 
> new comments have been addressed.

After another thought, seems we forgot to consider the alter table properties action.  I'm not sure whether it works for changing Iceberg table provider,  could you add `provider` property as `stringImmutablePropertyEntry` to `IcebergTablePropertiesMetadata#propertyEntries` to make it immutable?

**Comment by caican00** [comment]
> > > LGTM, except one minor comment
> > 
> > 
> > new comments have been addressed.
> 
> After another thought, seems we forgot to consider the alter table properties action. I'm not sure whether it works for changing Iceberg table provider, could you add `provider` property as `stringImmutablePropertyEntry` to `IcebergTablePropertiesMetadata#propertyEntries` to make it immutable?

ok, i will do this.
because iceberg supports modifying the table file format and users can modify the table file format in `Altertable` command using `write.format.default` instead of `provider`.

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@caican00 merged to main, thanks for your work

**Pull request merged**